File: java/client/src/main/java/glide/api/models/commands/bitmap/BitwiseOperation.java
/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.commands.bitmap;

import glide.api.commands.BitmapBaseCommands;

/**
 * Defines bitwise operation for {@link BitmapBaseCommands#bitop(BitwiseOperation, String,
 * String[])}. Specifies bitwise operation to perform between keys.
 *
 * @see <a href="https://redis.io/commands/bitop/">redis.io</a>
 */
public enum BitwiseOperation {
    AND,
    OR,
    XOR,
    NOT
}

File: node/src/commands/BitwiseOperation.ts
/**
 * Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
 */

// Import below added to fix up the TSdoc link, but eslint complains about unused import.
/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
import { BaseClient } from "src/BaseClient";

/**
 * Enumeration defining the bitwise operation to use in the {@link BaseClient.bitop|bitop} command. Specifies the
 * bitwise operation to perform between the passed in keys.
 */
export enum BitwiseOperation {
    AND = "AND",
    OR = "OR",
    XOR = "XOR",
    NOT = "NOT",
}
